libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -67,16 +70,36 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     return this.config.flavor === 'drawer';
   }
 
+  willClose$ = from(this.willClose).pipe(first());
+  private navigationEndListener$: any = this.router.events.pipe(
+    takeUntil(this.willClose$),
+    filter((event) => event instanceof NavigationEnd)
+  );
+
```

Comment:
So:
`willClose` is a Promise created from `ionModalWillDismiss` which is an Observable (Subject).
`willClose$` is an Observable created from the Promise `willClose` using `from()`
So maybe just:
```suggestion
  private navigationEndListener$: any = this.router.events.pipe(
    takeUntil(this.ionModalWillDismiss),
    filter((event) => event instanceof NavigationEnd)
  );

```